### PR TITLE
Generate reference page for /acs/systems/list

### PR DESCRIPTION
**Pull request by mastafit**
- Test v1.0
- Test v1.1
- Test v1.2
- Update types.ts
This is a snippet of a possible structure of a future docs generator, everything is adjustable

### CODE REVIEW

File: docs/scripts/generate-docs.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
+const fs = require("fs")
+const ejs = require("ejs")
+const path = require("path")
+
+// would handle data soon
+const data = {
+  objectName: "GetClientSession",
```
Comment (andrii-balitskyi) [inline docs/scripts/generate-docs.ts:7]: Could you explain what object you're referring to here? `GetClientSession` name sounds more like a method name.

File: docs/scripts/generate-docs.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  objectDescription: "Get a specified client session.",
+  methods: [
+    {
+      methodName: "GetClientSession",
```
Comment (andrii-balitskyi) [inline docs/scripts/generate-docs.ts:11]: Where is this pascal method name `GetClientSession` is used in the docs page? Do we need it in such a format?

Comment (mastafit) [reply docs/scripts/generate-docs.ts:11]: No, it's just a snippet to understand if we can do such transformation

File: docs/scripts/generate-docs.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+      description:
+        "Returns a specified [client session](../../core-concepts/authentication/client-session-tokens/).",
+      swaggerSrc: "https://connect.getseam.com/openapi.json",
+      swaggerPath: "/client_sessions/get",
+      swaggerMethod: "post",
```
Comment (andrii-balitskyi) [inline docs/scripts/generate-docs.ts:16]: I think Evan or Debbie mentioned that we'd move away from that swagger component we currently have on our pages. I'll ask Debbie on docs channel.
I think path and method would still be useful so we can just rename them.

File: docs/scripts/generate-docs.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    "client_session_id": "c75d4330-ae01-4dfd-b6c5-f3e94e0d8168",
+    "device_count": 4,
+    "connected_account_ids": [
+      "3ea0b67f-649f-4131-bfe3-f2035e77a3f9",
+      "6e1cad57-b244-40ca-b4f3-30a46c8000d4"
+    ],
+    "connect_webview_ids": [],
+    "user_identity_ids": []
+  }`,
+    },
```
Comment (andrii-balitskyi) [inline docs/scripts/generate-docs.ts:89]: I think we decided to leave code snippet generation out of the scope for the initial stage of docs generation.

File: docs/scripts/output.md
Comment (andrii-balitskyi) [inline docs/scripts/output.md:1]: It seems like the output file didn't pick up the mock data.

File: docs/scripts/generate-docs.ts (hunk 4 above)
Comment (mastafit) [reply docs/scripts/generate-docs.ts:89]: What do you mean by "leave out of scope"?

Comment (andrii-balitskyi) [reply docs/scripts/generate-docs.ts:89]: We won't implement code snippet generation for now.

File: docs/.DS_Store
Comment (andrii-balitskyi) [inline docs/.DS_Store:1]: You can put macOS-specific stuff in a global .gitignore file so they don't sneak into your commits.
https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer
https://github.com/github/gitignore/blob/main/Global/macOS.gitignore

**Comment by razor-x** [comment]
I just realized if we want to be able to preview these on gitbook we need to generate it to `./docs`. This should be ok as if we don't have any links to these pages no one should find them right? 🤔

**Comment by mastafit** [comment]
> I just realized if we want to be able to preview these on gitbook we need to generate it to `./docs`. This should be ok as if we don't have any links to these pages no one should find them right? 🤔 

I suppose it should be right

**Comment by razor-x** [comment]
@mastafit can you update the output path to docs so the gitbook preview works? Then we can see how close we are since the gitbook blocks will render

File: src/layouts/api-reference.hbs [hunk 5]
```diff
@@ -1,3 +1,104 @@
...
+
+# {{title}}
+
+{{{description}}}
+
+{% swagger src="https://connect.getseam.com/openapi.json" path="{{path}}"
+method="{{request.preferredMethod}}" %}
+[https://connect.getseam.com/openapi.json](https://connect.getseam.com/openapi.json)
+{% endswagger %} 
+
```
Comment (razor-x) [inline src/layouts/api-reference.hbs:13]:
```suggestion
```

**Review by razor-x — APPROVED** [review]

**Pull request merged**